app/addons/documents/index-results/components/results/ResultsScreen.js

```diff
@@ -0,0 +1,132 @@
+
+import React from 'react';
+import FauxtonAPI from '../../../../../core/api';
+import Constants from '../../../constants';
+import Components from "../../../../components/react-components";
+import {ResultsToolBar} from "../../../components/results-toolbar";
+import NoResultsScreen from './NoResultsScreen';
+import TableView from './TableView';
+
+const { LoadLines, Document } = Components;
+
+export default class ResultsScreen extends React.Component {
+  constructor (props) {
+    super(props);
+  }
+
+  componentDidMount () {
+    prettyPrint();
+  }
+
+  componentDidUpdate () {
+    prettyPrint();
+  }
+
+  onClick (id, doc) {
+    FauxtonAPI.navigate(doc.url);
+  }
+
+  getUrlFragment (url) {
+    if (!this.props.isEditable) {
+      return null;
+    }
+
+    return (
+      <a href={url}>
+        <i className="fonticon-pencil"></i>
+      </a>);
+  }
+
+  getDocumentList () {
+    let noop = () => {};
+    let data = this.props.results.results;
+
+    return _.map(data, function (doc, i) {
+      return (
+       <Document
+         key={doc.id + i}
+         doc={doc}
+         onClick={this.props.isEditable ? this.onClick : noop}
+         keylabel={doc.keylabel}
+         docContent={doc.content}
+         checked={this.props.isSelected(doc.id)}
+         header={doc.header}
+         docChecked={this.props.docChecked}
+         isDeletable={doc.isDeletable}
+         docIdentifier={doc.id} >
+         {doc.url ? this.getUrlFragment('#' + doc.url) : doc.url}
+       </Document>
+     );
+    }, this);
+  }
+
+  getDocumentStyleView () {
+    let classNames = 'view';
+
+    if (this.props.isListDeletable) {
+      classNames += ' show-select';
+    }
+
+    return (
+      <div className={classNames}>
+        <div id="doc-list">
+          {this.getDocumentList()}
+        </div>
+      </div>
+    );
+  }
+
+  getTableStyleView () {
+    return (
+      <div>
+        <TableView
+          onClick={this.onClick}
+          docChecked={this.props.docChecked}
+          isSelected={this.props.isSelected}
+          isListDeletable={this.props.isListDeletable}
+          data={this.props.results}
+          isLoading={this.props.isLoading}
+
+          removeItem={this.props.removeItem}
+          isChecked={this.props.allDocumentsSelected}
+          hasSelectedItem={this.props.hasSelectedItem}
+          toggleSelect={this.toggleSelectAll}
+          changeField={this.props.changeTableHeaderAttribute}
+          title="Select all docs that can be..." />
+      </div>
+    );
+  }
+
+  render () {
+    let mainView = null;
+
+    if (this.props.isLoading) {
+      mainView = <div className="loading-lines-wrapper"><LoadLines /></div>;
+    } else if (!this.props.hasResults) {
+      mainView = <NoResultsScreen text={this.props.textEmptyIndex}/>;
+    } else if (this.props.selectedLayout === Constants.LAYOUT_ORIENTATION.JSON) {
+      mainView = this.getDocumentStyleView();
+    } else {
+      mainView = this.getTableStyleView();
```

Comment:
I'd prefer to leave this as is.  By initializing `mainView` to `this.getTableStyleView()`, we do extra calculations when the end result is not the table view.